rasa_core/policies/ensemble.py

```diff
@@ -208,9 +208,29 @@ def probabilities_using_best_policy(self, tracker, domain):
                 max_confidence = confidence
                 result = probabilities
                 best_policy_name = 'policy_{}_{}'.format(i, type(p).__name__)
+
+        policy_names = [type(p).__name__ for p in self.policies]
+
+        # Trigger the fallback policy when ActionListen is predicted after
+        # a user utterance. This is done on the condition that: a fallback
+        # policy is present, there was just a user message and the predicted
+        # action is action_listen by a policy other than the MemoizationPolicy
+
+        if "FallbackPolicy" in policy_names:
+            idx = policy_names.index("FallbackPolicy")
+            fallback_policy = self.policies[idx]
+
+            if (result.index(max_confidence) == 0 and
+                not best_policy_name.endswith("MemoizationPolicy") and
```

Comment:
same with `MemoizationPolicy `

Reply:
what if this is the augmented memoization policy?

Reply:
let's make that explicit:
```
best_policy_name.endswith("." + MemoizationPolicy.__name__) or best_policy_name.endswith("." + AugmentedMemoizationPolicy.__name__)
``` 
and probably best to put the whole if condition in a separate function

Reply:
I mean they both end with "MemoizationPolicy" :D but yeah this makes sense if we rename it in future

Reply:
also do you mean the whole if statement or just the augmentedmemo/memoization policy one?

Reply:
whatever you think works better, the if should just be easier to read

Reply:
ok i think i'll just do the augmented/memo policy one